lamindb/examples/cellxgene.py

```diff
@@ -0,0 +1,246 @@
+
+
+def save_cxg_defaults() -> None:
+    """Add additional values from CellxGene schema to the instance.
+
+    CELLxGENE schemas use specific (control) values that are not available in the ontologies.
+    Therefore, we save them to the instance.
+    """
+    import bionty as bt
+
+    # "normal" in Disease
+    normal = bt.Phenotype.from_source(
+        ontology_id="PATO:0000461",
+        source=bt.Source.get(name="pato", currently_used=True),
+    )
+    bt.Disease(
+        uid=normal.uid,
+        name=normal.name,
+        ontology_id=normal.ontology_id,
+        description=normal.description,
+        source=normal.source,  # not sure
+    ).save()
+
+    # na, unknown
+    for model, name in zip(
+        [
+            bt.Ethnicity,
+            bt.Ethnicity,
+            bt.DevelopmentalStage,
+            bt.Phenotype,
+            bt.CellType,
+        ],
+        ["na", "unknown", "unknown", "unknown", "unknown"],
+    ):
+        model(ontology_id=name, name=name, description="From CellxGene schema.").save()
+
+    # tissue_type
+    tissue_type = ULabel(
+        name="TissueType",
+        is_type=True,
+        description='From CellxGene schema. Is "tissue", "organoid", or "cell culture".',
+    ).save()
+    for name in ["tissue", "organoid", "cell culture"]:
+        ULabel(name=name, type=tissue_type, description="From CellxGene schema.").save()
+
+    # suspension_type
+    suspension_type = ULabel(
+        name="SuspensionType",
+        is_type=True,
+        description='From CellxGene schema. This MUST be "cell", "nucleus", or "na".',
+    ).save()
+    for name in ["cell", "nucleus", "na"]:
+        ULabel(
+            name=name, type=suspension_type, description="From CellxGene schema."
+        ).save()
+
+
+def _create_cxg_sources(
+    categoricals: dict[str, FieldAttr], schema_version: str, organism: str
+) -> dict[str, SQLRecord]:
+    """Create a source dictionary of CELLxGENE categoricals to Source."""
+    import bionty as bt
+
+    def _fetch_bionty_source(entity: str, organism: str) -> SQLRecord | None:  # type: ignore
+        """Fetch the Bionty source of the pinned ontology."""
+        entity_sources = sources_df.loc[(sources_df.entity == entity)].copy()
+        if not entity_sources.empty:
+            if len(entity_sources) == 1:
+                row = entity_sources.iloc[0]  # for sources with organism "all"
+            else:
+                row = entity_sources[entity_sources.organism == organism].iloc[0]
+            source = bt.Source.filter(
+                organism=row.organism,
+                entity=f"bionty.{entity}",
+                name=row.source,
+                version=row.version,
+            ).one_or_none()
+            # if the source was not found, we register it from bionty-assets
+            if source is None:
+                source = getattr(bt, entity).add_source(
+                    bt.Source.using("laminlabs/bionty-assets")
```

Comment:
One shouldn't rely on `laminlabs/bionty-assets` as `add_source` can work with any sources, and you don't need them to be pre-curated in bionty-assets.

Simply do `add_source(source=<str>, version=..., organism=...)`.